src/Features/Core/Portable/ImplementInterface/AbstractImplementInterfaceService.CodeAction_Method.cs

```diff
@@ -25,14 +83,20 @@ internal partial class ImplementInterfaceCodeAction
                 bool useExplicitInterfaceSymbol,
                 string memberName)
             {
-                var syntaxFacts = Document.GetLanguageService<ISyntaxFactsService>();
+                var syntaxFacts = Document.GetRequiredLanguageService<ISyntaxFactsService>();
 
                 var updatedMethod = method.EnsureNonConflictingNames(State.ClassOrStructType, syntaxFacts);
 
                 updatedMethod = updatedMethod.RemoveInaccessibleAttributesAndAttributesOfTypes(
                     State.ClassOrStructType,
                     AttributesToRemove(compilation));
 
+                // Operators parameter should match containing type. For example, implementing:
+                // interface I { static abstract int operator -(I x); }
+                // in a class called C should result in:
+                // class C : I { public static int operator -(C x) {}
```

Comment:
```suggestion
                // class C : I { public static int operator -(C x) { } }
```

Reply:
also, this is a bit interesting from a covariant parameter type perspective.   It honestly seems a bit odd.  Given this, i could have a binary operator like `public static I operator +(I i1, I i2)`.  However, we'll instantiate that as `public static C operator +(C i1, C i2)`.  

Would i then be able to call this with two potentially different I's that were not both C's at runtmie?